src/NServiceBus.Core/Unicast/ContextualBus.cs

```diff
@@ -198,9 +194,27 @@ public void Reply<T>(Action<T> messageConstructor, NServiceBus.ReplyOptions opti
         /// <summary>
         /// <see cref="IBus.Reply"/>
         /// </summary>
-        public void Reply(object message,NServiceBus.ReplyOptions options)
+        public void Reply(object message, NServiceBus.ReplyOptions options)
         {
-            SendMessage(message.GetType(), message, options);
+            var pipeline = new PipelineBase<OutgoingReplyContext>(builder, settings, settings.Get<PipelineConfiguration>().MainPipeline);
+
+            var headers = new Dictionary<string, string>();
+
+
+            //todo: move to routing
```

Comment:
if this todo is a note to the next person i would have no idea what todo with it. can it be captured in an issue instead?